parser/options.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parser
+
+type options struct {
+	maxRecursionDepth  int
+	errorRecoveryLimit int
+	macros             map[string]Macro
+}
```

Comment:
Consider just using the fields on the `parser` and switching the `Option` to a `ParserOption`

e.g. `type ParserOption func(*parser) (*parser, error)`

Since the `ParserOption` implementations refer to a private type `parser` that means that they must all be curated by us, so I don't think we need a separate `options` object.

Reply:
The problem with parser is that it is not goroutine safe. I created Parser to have a re-usable interface which underneath creates new parser for each invocation, but doesn't need to build the map again.

I created options to keep the type private, instead of doing `func(*Parser)`.

Also ParserOption is repetative given that the package is parser.

Reply:
Fair point. I was trying to think about hoisting the options higher up. Ultimately, I'd like to have a `limits` package with all limits documented and configured in a single object. We can do this as a separate PR, but the idea would to address all compute / memory limits in a single place for the sake of getting a clear view of how to constrain CEL in a single place.  What do you think? I had #420 in mind as I wrote. I'm open to suggestions.

Reply:
That seems fine to me.